### PR TITLE
Add giflib

### PR DESCRIPTION
**Pull request by viniul**
This PR:

    Adds the giflib library to oss-fuzz (CC #402)
    Adds target out-of-source to seek feedback from maintainer team

CC @bshastry 
CC @eric-s-raymond

### CODE REVIEW

File: projects/giflib/project.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+homepage: "http://giflib.sourceforge.net/"
+primary_contact: "esr@thyrsus.com"
```
Comment (Dor1s) [inline projects/giflib/project.yaml]: Does this person have a GitHub account? Can you @mention them please to get their consent on the pull request?

File: projects/giflib/project.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+auto_ccs:
+  - "vincent.ulitzsch@live.de"
+sanitizers:
```
Comment (Dor1s) [inline projects/giflib/project.yaml]: I think you can use all sanitizers, the target should be working fine, just remove lines 5-6.

Comment (Dor1s) [reply projects/giflib/project.yaml]: Unless you have reasons to use ASan only.

File: projects/giflib/dgif_target.options [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[libfuzzer]
+close_fd_mask = 3
```
Comment (Dor1s) [inline projects/giflib/dgif_target.options]: Is this needed? I've just run the fuzz target and it doesn't seem to log anything.

**Comment by bshastry** [comment]
All good with me, go ahead with the merge.

**Comment by Dor1s** [comment]
There is an issue with CLA now. You've authored commits using 3 different emails:

```
$ git log -n 4 | egrep Author | uniq
Author: Vincent Ulitzsch <vincent.ulitzsch@live.de>
Author: Bhargava Shastry <bshastry@sect.tu-berlin.de>
Author: Vincent Ulitzsch <vincent@srlabs.de>
```

I'm afraid the CLA needs to be signed for all of these. Alternatively, we can close this PR and you'll re-upload it once again using the email that has CLA signed for. Sorry for the inconvenience.

**Comment by viniul** [comment]
I think I now signed the missing cla. @Dor1s can you confirm?

**Comment by Dor1s** [comment]
Just verified manually, the CLA is indeed signed for all the emails involved. Thanks everyone! Merging this now, since the CLA bot won't post an update here.

**Comment by kcc** [comment]
Looking at this fuzz target: it seems to OOM almost instantly: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=12518

Until the OOM is fixed, the target is unlikely to find anything interesting.

**Comment by kcc** [comment]
Also, I strongly recommend to rewrite this target to avoid any disk IO. 
This will not only make the target faster, but will unlock some of the fuzzing potential
(libFuzzer can lose some knowledge when you go through IO like this).

I think you can implement what's needed using `DGifOpen`

**Pull request merged**